test/suites/metrics.sh

```diff
@@ -10,9 +10,11 @@ test_metrics() {
   # c1 metrics should show as the container is running
   lxc query "/1.0/metrics" | grep "name=\"c1\""
 
-  # c2 metrics should show the container as stopped
-  lxc query "/1.0/metrics" | grep "name=\"c2\""
-  lxc query "/1.0/metrics" | grep "name=\"c2\"" | grep "state=\"STOPPED\""
+  # c2 metrics should not exist as it's not running
+  ! lxc query "/1.0/metrics" | grep "name=\"c2\"" || false
+
+  # Check that we can get the count of existing containers. There should be two: c1 (RUNNING) and c2 (STOPPED).
+  lxc query /1.0/metrics | grep -E '^(lxd_containers)\{.*\} 2' || false
```

Comment:
```suggestion
  lxc query /1.0/metrics | grep -xE 'lxd_containers\{project="default"\} 2'
```
Using `-x` here to have full line matching and project label added. I also removed the `|| false` which seemed like a copy-n-pasto.

Reply:
I'd also validate the `lxd_vms` metric's presence and value of `0`.

Reply:
fixed